### ISSUE TITLE
Adapt the sharee searching to the capability "min number of characters to search"

### ISSUE THREAD

**Issue by mbode28** [owncloud/android#2300]
### Actual behaviour
When typing a user name to share an item with, no user is found, if username is shorter than 4 characters

### Expected behaviour
After typing a couple of characters of the user name, a selection of matching user names should be displayed like it is in the web frontend or the KDE client.
 
### Steps to reproduce
1. Select a file in owncloud app (long press)
2. tap sharing icon
3. tap '+' sign on "Nutzer und Gruppen" (users and groups)
4. type username of another user on the owncloud server which has only 2 characters. Example: ms

Can this problem be reproduced with the official owncloud server?
(url: https://demo.owncloud.org, user: test, password: test)

Don't know if there are short usernames.

### Environment data
Android version: 8.0.0

Device model: Samsung Galaxy S9

Stock or customized system: stock

ownCloud app version: 2.8.0

ownCloud server version: 10.0.10
### Related info
- [Admin manual](https://doc.owncloud.com/server/admin_manual/configuration/server/config_sample_php_parameters.html#defines-the-minimum-characters-entered-before-a-search-returns-results-for)

### PR
- App: #2678 
- Library: https://github.com/owncloud/android-library/pull/270

**Comment by michaelstingl** [comment]
@mbode28 minimum required characters can be configured on the server:
https://github.com/owncloud/core/blob/v10.0.10/config/config.sample.php#L282-L290

Please check if this fixes your problem.

**Comment by mbode28** [comment]
This setting makes no difference in the android app. It make a difference
in the web frontend though.

Am Mo., 1. Okt. 2018 um 16:14 Uhr schrieb Michael Stingl <
notifications@github.com>:

**Comment by michaelstingl** [comment]
@davigonz @jesmrec Could you have a look?

**Comment by jesmrec** [comment]
we changed this behaviour time ago to avoid a request flooding that makes servers with a huge number of users to run very very slow while sharees are being searched.

A "workaround" is  typing a blank space after the two characters.

**Comment by mbode28** [comment]
The trick with the spaces doesn't work for me. But I found another one:
changed the full name of the user to something longer than the username and
then enter the full name.

For me that's ok. You may close the bug.

Thanks for your very fast help.

Jesús Recio Rincón <notifications@github.com> schrieb am Mo., 1. Okt. 2018,
16:47:

**Comment by jesmrec** [comment]
the trick works by completing till 4 characters, in your case the username has 2 characters, so you need 2 blanks. But it is cool your solution as well. Thanks for reporting, if you need something else related with the issue, please reopen.

**Comment by michaelstingl** [comment]
Capabilities could be read from the server:
`OC.getCapabilities().files_sharing.search_min_length`

or as XML
```
<files_sharing>
<search_min_length>2</search_min_length>
</files_sharing>
```

**Comment by jesmrec** [comment]
so we can adapt the apps to the capability. reopening...

**Comment by davigonz** [comment]
Done, to be included in 2.14
